stringscore.go

```diff
@@ -25,16 +25,16 @@ func Score(target string, query string) int {
 		return 0 // impossible for query to be a substring
 	}
 
-	targetLower := strings.ToLower(target)
-	queryLower := strings.ToLower(query)
+	targetRunes := []rune(target)
+	queryRunes := []rune(query)
+	targetLower := []rune(strings.ToLower(target))
+	queryLower := []rune(strings.ToLower(query))
 
-	startAt := 0
 	score := 0
 
-	for queryIdx := 0; queryIdx < len(query); queryIdx++ {
-		targetIdx := strings.IndexByte(targetLower[startAt:], queryLower[queryIdx]) + startAt
-
-		if targetIdx < startAt {
+	for queryIdx := 0; queryIdx < len(queryRunes); queryIdx++ {
+		targetIdx := runeIndex(targetLower, queryLower[queryIdx])
```

Comment:
@slimsag I know you followed up with a fix around here, but you actually changed the scores. You now award the start of word bonus more often, since targetIdx can be zero not only at the beginning of target. If you look at the original line `targetIdx := find(...[startAt:], ...) + startAt`.

Reply:
Fixed in #5